### DISCUSSION TITLE
Autocompleting some strings should place a StringName literal instead of a normal string

### DISCUSSION THREAD

**Discussion by jaymehta-g** [godotengine/godot-proposals#6620] [Scripting]
If I have
`Input.is_action_pressed("ui_left")`
running in `_process` then if I understand correctly, every single frame it is creating a new String object, comparing it to all currently held input keys letter by letter until it finds a match. But if I run
`Input.is_action_pressed(&"ui_left")`
, then Godot turns the string into a reference to the same StringName object and every frame the line is much faster. 
The problem is that when typing out this line, if I use the auto complete options in the dropdown that appears while I'm typing the name of the input axis, Godot inserts a normal string when it would make much more sense to insert a StringName and be more efficient.

**Comment by dalexeev** [comment]
This is implemented as 2 editor settings (disabled by default) for future 4.3. See:

* #5496
* godotengine/godot#66481
